src/config.rs

```diff
@@ -425,29 +425,84 @@ pub struct HtmlConfig {
     pub livereload_url: Option<String>,
     /// Should section labels be rendered?
     pub no_section_label: bool,
+    /// Search settings.
+    pub search: Search,
 }
 
 /// Configuration for tweaking how the the HTML renderer handles the playpen.
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
 #[serde(default, rename_all = "kebab-case")]
 pub struct Playpen {
-    /// The path to the editor to use. Defaults to the [Ace Editor].
-    /// 
+    /// The path to the editor to use. If not specified, will use the [Ace Editor] implementation
+    /// included in MDBook. Default: `""`.
+    ///
     /// [Ace Editor]: https://ace.c9.io/
     pub editor: PathBuf,
-    /// Should playpen snippets be editable? Defaults to `false`.
+    /// Should playpen snippets be editable? Default: `false`.
     pub editable: bool,
 }
 
 impl Default for Playpen {
     fn default() -> Playpen {
         Playpen {
-            editor: PathBuf::from("ace"),
+            editor: PathBuf::new(),
             editable: false,
         }
     }
 }
 
+/// Configuration of the search functionality of the HTML renderer.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(default, rename_all = "kebab-case")]
+pub struct Search {
+    /// Enable in browser searching. Default: `true` (if `search` feature is enabled).
+    pub enable: bool,
+    /// The path to the searcher to use. If not specified, will use the searcher code included in
+    /// MDBook. Default: `""`
+    pub searcher: PathBuf,
```

Comment:
Hmm, I think we may end up having a similar issue to themes here. By adding this to the config file we're *theoretically* allowing people to swap in their own implementations, but because there is no well defined (and documented) interface to our `searcher`, people probably won't be able to make use of this knob in practice.

What would one gain by being able to use their own `searcher` implementations?

Reply:
I agree that the way it's setup is not ideal, I just copied the setup used for playpen_editor.

While we give no guarantees about the stability or interfaces for the searcher, the current system can be useful in practice for making small changes to search behavior (remove highlighting, change keyboard shortcuts, etc.). I'll agree, however, that this feature is unlikely to be used much if at all. In general I don't think many users will make significant theme changes out-of-tree until we provide a better way to do that.

So what change can we make to this PR? Here's what I'm thinking:

1. Remove the `editor` and `searcher` settings and the corresponding file override behavior.
2. Add `copy-js` settings for both. If set to false, the JS code for the feature will not be copied to the output directory. This is more flexible since users can then include whatever JS they want with `additional-js`, and it reduces the complexity of this little-used feature to almost zero.

Reply:
I think that sounds quite reasonable :+1:

Besides, it's not hard to put back in if we find people actually want the feature.